src/adapter/scriptSkipper.ts

```diff
@@ -0,0 +1,86 @@
+import cdp from '../cdp/api';
+
+export class ScriptSkipper {
+  private _userSkipPatterns: string[];
+  private _nonNodeInternalRegex: string = "";
+
+  // filtering node internals
+  private _nodeInternalsRegex: string = "";
+  skipAllNodeInternals: boolean = false;
+
+  private _isUrlSkippedMap = new Map<string, boolean>();
+
+  private _blackboxedUrls: string[] = [];
+  private blackboxSender?: (params: cdp.Debugger.SetBlackboxPatternsParams) => Promise<cdp.Debugger.SetBlackboxPatternsResult | undefined>;
+
+  constructor(skipPatterns: string[]) {
+    this._userSkipPatterns = skipPatterns;
+    this._preprocessNodeInternals();
+    this._setRegexForNonNodeInternals();
+  }
+
+  private _preprocessNodeInternals(): void {
+    let nodeInternalRegex = new RegExp("^<node_internals>\/(.*)$");
```

Comment:
You don't have to use the RegExp constructor, you can use a regex literal (like `/^<node_internals>/.*`/)